packages/open-next/src/adapters/server-adapter.ts

```diff
@@ -0,0 +1,262 @@
+      return rawQueryString.length > 0
+        ? `${rawPath}?${rawQueryString}`
+        : rawPath;
+    },
+    get body() {
+      const { body, isBase64Encoded } = event;
+      if (Buffer.isBuffer(body)) {
+        return body;
+      } else if (typeof body === "string") {
+        return Buffer.from(body, isBase64Encoded ? "base64" : "utf8");
+      } else if (typeof body === "object") {
+        return Buffer.from(JSON.stringify(body));
+      }
+      return Buffer.from("", "utf8");
+    },
+    get headers() {
+      const { headers: rawHeaders, cookies } = event;
+
+      const headers: Record<string, string> = {};
+
+      if (Array.isArray(cookies)) {
+        headers["cookie"] = cookies.join("; ");
+      }
+
+      for (const [key, value] of Object.entries(rawHeaders || {})) {
+        headers[key.toLowerCase()] = value!;
+      }
+
+      return headers;
+    },
+    get remoteAddress() {
+      return event.requestContext.http.sourceIp;
+    },
+  }),
+  cloudfront: (event: CloudFrontRequestEvent) => ({
+    get method() {
+      return event.Records[0].cf.request.method;
+    },
+    get rawPath() {
+      return event.Records[0].cf.request.uri;
+    },
+    get url() {
+      const { uri, querystring } = event.Records[0].cf.request;
+      return querystring.length > 0 ? `${uri}?${querystring}` : uri;
+    },
+    get body() {
+      const { body } = event.Records[0].cf.request;
+      if (!body) {
+        return Buffer.from("", "utf8");
+      }
+
+      return body.encoding === "base64"
+        ? Buffer.from(body.data, "base64")
+        : Buffer.from(body.data, "utf8");
+    },
+    get headers() {
+      const { headers: rawHeaders } = event.Records[0].cf.request;
+      const headers: Record<string, string> = {};
+
+      for (const [key, values] of Object.entries(rawHeaders)) {
+        for (const { value } of values) {
+          if (value) {
+            headers[key] = value;
+          }
+        }
+      }
+
+      return headers;
+    },
+    get remoteAddress() {
+      return event.Records[0].cf.request.clientIp;
+    },
+  }),
+};
+
+/////////////
+// Handler //
+/////////////
+
+export async function handler(
+  event: APIGatewayProxyEventV2 | CloudFrontRequestEvent
+) {
+  debug(event);
+
+  // Parse Lambda event and create Next.js request
+  const isCloudFrontEvent = (event as CloudFrontRequestEvent).Records?.[0]?.cf;
+  const parser = isCloudFrontEvent
+    ? eventParser.cloudfront(event as CloudFrontRequestEvent)
+    : eventParser.apiv2(event as APIGatewayProxyEventV2);
+  const req = new IncomingMessage({
+    method: parser.method,
+    url: parser.url,
+    headers: parser.headers,
+    body: parser.body,
+    remoteAddress: parser.remoteAddress,
+  });
+  const res = new ServerResponse({ method: req.method });
+
+  // Process Next.js request
+  await processRequest(req, res);
+
+  // Format Next.js response to Lambda response
+  const statusCode = res.statusCode;
```

Comment:
```suggestion
  const statusCode = res.statusCode ?? 200;
```

Nextjs app dir route does not return status code for some reason (it only returns status code of the root `app/route.ts`)